File: LatinReader/app/src/main/java/com/ericmschmidt/latinreader/utilities/ITextConverter.java
package com.ericmschmidt.latinreader.utilities;

import android.text.TextWatcher;
import android.widget.EditText;

/** Interface for converter that changes one orthography for another.
 *
 * @author Eric Schmidt
 * @author http://telpirion.com
 * @version 1.5
 * @since 1.1
 */
public interface ITextConverter {
    public String convertSourceToTargetCharacters(String source);
    public String convertTargetToSourceCharacters(String target);
    public String getLang();
    public TextWatcher getTextWatcher(EditText editText);
}

File: LatinReader/core-data/src/main/java/com/telpirion/classicsreader/core_data/utilities/ITextConverter.kt
package com.telpirion.classicsreader.core_data.utilities

/**
 * Interface for a converter that changes one orthography for another.
 *
 * @author Eric Schmidt
 * @author http://telpirion.com
 * @version 1.5
 * @since 1.1
 */
interface ITextConverter {
    fun convertSourceToTargetCharacters(source: String): String
    fun convertTargetToSourceCharacters(target: String): String
    val lang: String
}
